tests/serdetest.c

```diff
@@ -48,4 +49,213 @@ TEST (musig2, commitments_serialize_deserialize) {
         ASSERT_EQ(serde_batch_list[i], serialized_batch_list[i]);
 }
 
+/* Fuzz a public key then try to generate the aggregate public key */
+TEST (musig2, fuzz_pubkey_precomputation) {
+    secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_VERIFY);
+    musig2_context_signer mcs_list[NR_SIGNERS]; // Array that holds NR_SIGNERS musig2_context_signer
+    musig2_context_signature mps[NR_SIGNERS];
+    unsigned char serialized_batch_list[NR_MESSAGES * NR_SIGNERS * V * MUSIG2_PUBKEY_BYTES_COMPRESSED];
+    unsigned char serialized_pubkey_list[NR_SIGNERS * MUSIG2_PUBKEY_BYTES_COMPRESSED];    // Signers' public key list
+    unsigned char signature[MUSIG2_BYTES];
+    secp256k1_pubkey pubkey;
+    MUSIG2_ERROR res;
+    int err;
+
+    res = musig2_helper_setup(mcs_list, serialized_pubkey_list, serialized_batch_list, NR_SIGNERS);
+    ASSERT_EQ(res, MUSIG2_OK);
+
+    int fuzz_index = (rand() % NR_SIGNERS) * MUSIG2_PUBKEY_BYTES_COMPRESSED;
+    musig2_helper_fuzz_data(&serialized_pubkey_list[fuzz_index], MUSIG2_PUBKEY_BYTES_COMPRESSED);
+
+    res = musig2_helper_precomputation(serialized_pubkey_list, serialized_batch_list, mcs_list, NR_SIGNERS);
+
+    switch(res){
+        case MUSIG2_ERR_PARSE_PK:
+            err = secp256k1_ec_pubkey_parse(ctx, &pubkey, &serialized_pubkey_list[fuzz_index], MUSIG2_PUBKEY_BYTES_COMPRESSED);
+            ASSERT_EQ(err, 0);
+            break;
+        case MUSIG2_ERR_AGGR_PK:
+            err = secp256k1_ec_pubkey_parse(ctx, &pubkey, &serialized_pubkey_list[fuzz_index], MUSIG2_PUBKEY_BYTES_COMPRESSED);
+            ASSERT_EQ(err, 1);
+            break;
+        default:
+            err = secp256k1_ec_pubkey_parse(ctx, &pubkey, &serialized_pubkey_list[fuzz_index], MUSIG2_PUBKEY_BYTES_COMPRESSED);
+            ASSERT_EQ(err, 1);
+
+            err = musig2_helper_sign(mcs_list, mps, NR_SIGNERS);
+            ASSERT_EQ(err, MUSIG2_OK);
+
+            err = musig2_aggregate_partial_sig(mps, signature, NR_SIGNERS);
+            ASSERT_EQ(err, MUSIG2_OK);
+
+            err = musig2_helper_verify(serialized_pubkey_list, signature, MSG_1, MSG_1_LEN, NR_SIGNERS);
+            ASSERT_EQ(err, MUSIG2_INVALID);
```

Comment:
Here we are really only concerned about the serialisation of the public key. Not about the signature. We already check that in the test below. We can remove these checks, otherwise we have duplicate code/tests.
```suggestion
        default:
            err = secp256k1_ec_pubkey_parse(ctx, &pubkey, &serialized_pubkey_list[fuzz_index], MUSIG2_PUBKEY_BYTES_COMPRESSED);
            ASSERT_EQ(err, 1);
```